src/actions/undo.cpp

```diff
@@ -360,7 +360,7 @@ void undo_list::undo()
 	undos_.pop_back();
 	if (undo_action* undoable_action = dynamic_cast<undo_action*>(action.get()))
 	{
-		int last_unit_id = resources::gameboard->unit_id_manager().get_save_id();
+		int last_unit_id = static_cast<int>(resources::gameboard->unit_id_manager().get_save_id());
```

Comment:
If `get_save_id` and `set_save_id` use std::size_t, then why is this `int`?

Making it `std::size_t` would require fixing the sanity check comparing it to `undoable_action->unit_id_diff`, and that descends into a rabbit warren of the TODO comment that's on `undo_action::unit_id_diff`, however the cast to `int` doesn't look right.

Reply:
I'm not trying to make a judgement on whether it's correct or not - the PR is mainly just aimed at making instances like this more explicit in what they're doing.

Reply:
Making them explicit will suggest to anyone reviewing the code later that someone made the judgement, so either they'll skip the cleanup, or they'll spend time working with that apparent judgement.

These are just lint warnings - it's good to fix them and I welcome this PR opening the debate, but if we merge as-is then months down the line we'll be used to looking at code, wondering why it does that, running `git blame` and then mumbling "oh, it's that set of quick lint fixes".

Reply:
Added a new commit with an alternate change for this.